src/uid2DiiNormalization.ts

```diff
@@ -0,0 +1,78 @@
+enum EmailParsingState {
+  Starting,
+  SubDomain,
+}
+
+export type NormalizedPhone = string;
+export type NormalizedEmail = string;
+
+export function isNormalizedPhone(phone: string): phone is NormalizedPhone {
+  return /^\+[0-9]{10,15}$/.test(phone);
+}
+
+export function normalizeEmail(email: string): NormalizedEmail | undefined {
+  if (email == undefined || email.length <= 0) {
+    return undefined;
+  }
+
+  let preSb = "";
+  let preSbSpecialized = "";
+  let sb = "";
+  let wsBuffer = "";
```

Comment:
I'm finding it hard to understand what these variable names mean - could you please explain?